.gitignore

```diff
@@ -5,4 +5,5 @@
 .*.swp
 .history
 **/.vscode
-**/.venv
\ No newline at end of file
+**/.venv
```

Comment:
PR on the whole looks good. It will be nice if integration tests can be added.
ref - https://github.com/metacontroller/metacontroller/tree/master/test/integration/composite